### ISSUE TITLE
Checklist for 1.0.0 release

### ISSUE THREAD

**Issue by olgabot** [nf-core/kmermaid#29]
@phoenixAja Looks like this is a good overview: https://nf-co.re/developers/adding_pipelines#making-the-first-release

@nf-core/core Are there any other expectations we should be aware of? e.g. is a MultiQC report expected?

**Comment by phoenixAja** [comment]
@olgabot perfect thanks!

**Comment by apeltzer** [comment]
No, this is hopefully complete - if anything is unclear / missing in your opinion please comment here / ping core again 👍

**Comment by olgabot** [comment]
Thank you both!!

**Comment by ewels** [comment]
MultiQC reports are _nice_ of course 😉  but no, they're not a requirement.

The link you posted is good, also take a look at https://nf-co.re/pipeline_health - it looks like you're missing the `TEMPLATE` branch which is needed for synchronisation.

Though, arg, looks like you didn't start with the template 🙈 (in 33f94428e38541d80899117903b2ef94bd56389b). So you'll need to manually set up synchronisation - see https://nf-co.re/developers/sync

**Comment by phoenixAja** [comment]
@ewels yes this is our main problem! Thanks for the link!

**Comment by ewels** [comment]
Depending on how much you care about commit / contributor history, the _easiest_ way to do this is to just make a new pipeline from scratch and then copy + paste your new code in as a second commit. You can then force push to wipe everything in this repo. Very, very blunt as it essentially deletes everything. But it's the quickest way to get stuff working 😄 

The alternative of manual syncing isn't too bad otherwise, and you get a lot of experience in how to deal with resolving merge conflicts! Just be careful with reviewing the final solution to make sure that you didn't miss anything / make any mistakes, as it's pretty easy to do so..

**Comment by phoenixAja** [comment]
that idea did cross my mind. I'll try manually adding first, thankfully I have `git reflog` in case mistakes are made.

**Comment by olgabot** [comment]
@phoenixAja if it's a total mess we should check with other contributors @pranathivemuri and @snafees if they're okay with blasting out the repo. I'm okay with it as it's relatively early and it's likely they'll have future changes as well. But if we want to preserve the previous ones then `git reflog` and `cherry-pick` it is!

**Comment by pranathivemuri** [comment]
Outstanding PRs aside, it seems like the kmermaid repo has only 2 contributors, me and Olga currently! Ideally i would like to have the contributions I made in my name, but if that’s blocking a release, I wouldn’t mind if someone else copy pasted manually all the code.  also, i think there is a number of places where czbiohub/nf-kmer-similarity exists instead of nf-core/kmermaid and the docker container iname should be changed/synced to this repo as well.

**Comment by phoenixAja** [comment]
that's a different issue to fix @pranathivemuri so i wont tackle the docker stuff in this branch. I guess the commit history is already messed up with this repo (all of the `ska` stuff i wrote got overwritten, so I'm no longer an author on this pipeline).

**Comment by phoenixAja** [comment]
@ewels i have the `.git` directory in addition to the following files, should i remove it before proceeding? It seems wrong to delete that, but it's not included in the list of dirs/files below. I just wanted to double check.
```
Untracked files:
  (use "git add <file>..." to include in what will be committed)

  .gitattributes
  .github/
  .gitignore
  .travis.yml
  CHANGELOG.md
  CODE_OF_CONDUCT.md
  Dockerfile
  LICENSE
  README.md
  assets/
  bin/
  conf/
  docs/
  environment.yml
  main.nf
  nextflow.config
```

**Comment by pranathivemuri** [comment]
@phoenixAja ska stuff isn't merged into master yet. it is still in a PR as far as I can see - https://github.com/nf-core/kmermaid/pull/19/files 

The only ska stuff added in the nf-core are right now the dependencies in environment.yml files. we should review and merge/make a decision on outstanding PRs before we merge.

**Comment by phoenixAja** [comment]
still wierd that that got added to `environment.yml`.. Also yes of course we are definitely planning on reviewing before any PR is approved.

**Comment by phoenixAja** [comment]
> @ewels i have the `.git` directory in addition to the following files, should i remove it before proceeding? It seems wrong to delete that, but it's not included in the list of dirs/files below. I just wanted to double check.
> 
> ```
> Untracked files:
>   (use "git add <file>..." to include in what will be committed)
> 
>   .gitattributes
>   .github/
>   .gitignore
>   .travis.yml
>   CHANGELOG.md
>   CODE_OF_CONDUCT.md
>   Dockerfile
>   LICENSE
>   README.md
>   assets/
>   bin/
>   conf/
>   docs/
>   environment.yml
>   main.nf
>   nextflow.config
> ```

definitely need to keep `.git`..

**Comment by ewels** [comment]
If you take the nuclear option, you need to start in a totally fresh empty new directory. Then run `nf-core create`. It will initialise a new git repository (including a new `.git` directory) as part of that process. It will set up the required branches and make initial commits with the blank template.

Once that is complete, you can take your existing pipeline code and copy + paste it on top of the existing files (***not*** including the `.git` directory). You can then commit that. Git will do the difficult job of figuring out what you changed from the original template, facilitating synchronisation later on with new templates.

Once you've updated the new repo locally with your code, you can add a git remote for the existing GitHub repository and force push. If you do a regular push it will reject it as the histories are not compatible. If you do a force push then I think it will just wipe everything on GitHub and you'll have the two commits - one from the template and one with your pipeline. Make sure that you push all branches (`git push --all`).

See the docs about creating a new template for details: https://nf-co.re/developers/adding_pipelines#create-a-pipeline-from-the-template

Maybe you already knew all this stuff? 😀 Just wasn't quite sure what you had done to get to https://github.com/nf-core/kmermaid/issues/29#issuecomment-550426819 though, so thought it worth describing.

**Comment by pranathivemuri** [comment]
closing this as 1.0.0 is released
